### PR TITLE
Fix custom pulse instruction conversion to Qobj.

### LINKED ISSUE #11828: Converting custom pulse play instructions to Qobj raises an error
### Environment

- **Qiskit version**: 1.0
- **Python version**: 3.9
- **Operating system**: Windows

### What is happening?

When you try to convert to Qobj a circuit with a custom pulse in it, a `QiskitError` is raised.

### How can we reproduce the issue?

```python
import qiskit
from qiskit import pulse
from qiskit import circuit
from qiskit.providers.fake_provider import GenericBackendV2

with pulse.build() as sch:
    pulse.play(pulse.Triangle(32, 0.1, 0.2), pulse.DriveChannel(0))

qc = circuit.QuantumCircuit(1, 1)
qc.x(0)
qc.measure_all()
qc.add_calibration("x", (0,), sch)

transpiled = qiskit.transpile(qc, backend=GenericBackendV2(2))

qiskit.assemble(transpiled)
```
The pulse choice here is immaterial, as long as it is a pulse type which ~~is not recognized by the backend~~ does not appear in [ParametricPulseShapes](https://github.com/Qiskit/qiskit/blob/12f24a76b6fbba4ebd9d7cf09c42f2c6c6e3c8f7/qiskit/qobj/converters/pulse_instruction.py#L35). `Triangle` ~~is not recognized by the backend~~ does not appear there. ~~(Edit - `GenericBackendV2` doesn't define the field `parametric_pulses` so all pulse types will behave the same way here).~~

### What should happen?

The circuit should convert successfully to Qobj.

### Any suggestions?

PR #11410 changed the error raised in [ParametricPulseShapes](https://github.com/Qiskit/qiskit/blob/12f24a76b6fbba4ebd9d7cf09c42f2c6c6e3c8f7/qiskit/qobj/converters/pulse_instruction.py#L68) from a `ValueError` to a `QiskitError`. However, other modules relied on the `ValueError` to identify pulses not supported by the backend - [here](https://github.com/Qiskit/qiskit/blob/12f24a76b6fbba4ebd9d7cf09c42f2c6c6e3c8f7/qiskit/assembler/assemble_schedules.py#L233).

### PR DESCRIPTION
**Pull request by TsafrirA**
### Summary
This is a possible fix to #11828.

### Details and comments
The behavior of [ParametricPulseShapes](https://github.com/Qiskit/qiskit/blob/12f24a76b6fbba4ebd9d7cf09c42f2c6c6e3c8f7/qiskit/qobj/converters/pulse_instruction.py#L68) is restored to raise `ValueError` when the pulse shape is not supported, and `QiskitError` when the pulse object is not recognized. The need for the second scenario is not clear to me, but as a quick fix I reverted to the code before #11410.

If we think that the two error types are not needed, we can alternatively change the error type used by the [assemble logic](https://github.com/Qiskit/qiskit/blob/12f24a76b6fbba4ebd9d7cf09c42f2c6c6e3c8f7/qiskit/assembler/assemble_schedules.py#L233).

fixes #11828

### CODE REVIEW

**Comment by TsafrirA** [comment]
As a side note, I think this bug highlights some deficiencies in the tests. We should probably beef up the tests there, but I didn't want to tie it up with the fix.

File: releasenotes/notes/fix-custom-pulse-qobj-conversion-5d6041b36356cfd1.yaml [hunk 1]
```diff
@@ -0,0 +1,7 @@
+---
+fixes:
+  - |
+    Fixed a bug in the conversion of custom pulse instructions to ``Qobj``. The bug was introduced in Qiskit 1.0.0
+    and caused conversion of custom pulse instructions to raise an error. After the fix, the conversion is
+    carried out correctly, and the custom pulse is converted to ``Waveform`` as it should.
```
Comment (wshanks) [inline releasenotes/notes/fix-custom-pulse-qobj-conversion-5d6041b36356cfd1.yaml]:
```suggestion
    Fixed a bug in the conversion of custom pulse instructions to the legacy :mod:`qiskit.qobj` format. The bug was introduced in Qiskit 1.0.0
    and caused conversion of instructions with custom pulse shapes to raise an error. After the fix, the conversion is
    carried out correctly, and the custom pulse is converted to :class:`.Waveform` as it should.
```

**Review by wshanks — DISMISSED** [review]
Nice fix, @TsafrirA. I just had some wording suggestions for the release notes.

We are trying to move away from Qobj any way, but it might be worth adding a custom pulse shape to this test or one like it:

https://github.com/Qiskit/qiskit/blob/12f24a76b6fbba4ebd9d7cf09c42f2c6c6e3c8f7/test/python/compiler/test_assembler.py#L502-L519

**Comment by TsafrirA** [comment]
Thanks @wshanks! I added a test to cover something like this.

**Review by wshanks — APPROVED** [review]
Looks good!

**Review by nkanazawa1989 — APPROVED** [review]
Thanks for the quick fix!

**Pull request merged**